src/lua/uri.lua

```diff
@@ -49,6 +64,16 @@ local function parse(str)
             result[k] = ffi.string(uribuf[k])
         end
     end
+    result.params = {}
+    for param_idx = 0, uribuf.param_count - 1 do
+        local name = ffi.string(uribuf.params[param_idx].name)
+        result.params[name] = result.params[name] or {}
+        result.params[name].values = result.params[name].values or {}
```

Comment:
Let's drop `values` and store the query parameter values immediately in `params[name]`?

Reply:
fixed